### PR TITLE
Revert PR sync for gitlab

### LINKED ISSUE #460: Create revert pr sync for gitlab to map incidents
Create a service module to sync revert prs for gitlab. We need to create a mapping of reverted pr and original pr. This will be later used by the incident service to map incidents with revert prs.

### PR DESCRIPTION
**Pull request by adnanhashmi09**
## Linked Issue(s) 
Closes #460 

## Proposed Changes
A service module to sync gitlab revert prs. This will be called by the gitlab sync service which will be later implemented. Tests are also written for the same service.

### CODE REVIEW

File: backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+        repo_ids: Set[str] = set()
+        repo_id_to_pr_merge_hash_to_id_map: Dict[str, Dict[str, str]] = {}
+        pr_merge_hash_match_strings: List[str] = []
+
+        for pr in prs:
+            if pr.state != PullRequestState.MERGED:
+                continue
+
+            pr_merge_hash_match_strings.append(f"revert-{pr.merge_commit_sha[:8]}")
```
Comment (samad-yar-khan) [inline backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py:62]: is this pattern for revert pr branch names always true ?

File: backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        self, prs: List[PullRequest]
+    ) -> List[PullRequestRevertPRMapping]:
+        """
+        This function takes a list of PRs and for each PR it tries to
+        find if that pr is a revert PR and for that revert PR it tries
+        to find the original which was reverted and create a mapping.
+        """
+        revert_pr_hashes: List[str] = []
+        repo_ids: Set[str] = set()
+        repo_id_to_pr_merge_hash_to_id_map: Dict[str, Dict[str, str]] = {}
```
Comment (samad-yar-khan) [inline backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py:121]: repo_id_to_pr_merge_hash_to_revert_pr_id_map would make this more readable as this code is already kind of complex

File: backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py (hunk 1 above)
Comment (adnanhashmi09) [reply backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py:62]: yep

File: backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py (hunk 2 above)
Comment (adnanhashmi09) [reply backend/analytics_server/mhq/service/code/sync/revert_pr_gitlab_sync.py]: Yeah, makes sense. Made the changes. Thanks for the suggestion.

**Review by samad-yar-khan — APPROVED** [review]

**Pull request merged**